src/models/account.model.js

```diff
@@ -32,85 +35,108 @@ class Account {
+    return client.query(query)
+      .then(result => result.rows[0]);
+  }
+
+  static async create(ownerID, type, status, balance) {
+    const accountNumber = Account.genAccountNumber();
+    const createdOn = new Date();
+    const query = `INSERT INTO accounts(accountnumber, createdon, status, userid, type, balance) 
+    values($1, $2, $3, $4, $5, $6)
+    RETURNING accountnumber`;
+    const result = await client.query(query, [accountNumber, createdOn, status, ownerID, type, balance]);
+    return result.rows[0];
   }
 
   /**
   * credit an account
   * @param {string} req
   * @returns {Integer} returns account balance
   */
-  credit(val) {
-    let balance = Number(this.balance);
-    const debitAmt = Number(val);
+  async credit(amount) {
+    const { accountNumber } = account;
+    const account = await this.find('accountnumber', accountNumber);
+    let { balance } = account;
+    const debitAmt = Number(amount);
     balance += debitAmt;
-    this.balance = balance.toString();
-    return this.balance;
+    const query = `UPDATE accounts SET accountbalance = ${balance} WHERE accountnumber = ${accountNumber}`;
+    const result = await client.query(query);
+    return result.rows[0];
   }
 
   /**
   * Changes account status
   * @returns {string} returns account status
   */
-  toggleState() {
-    this.status = this.status === 'active' ? 'domant' : 'active';
-    return this.status;
-  }
-
-  /**
-  * Save account to datastore
-  * @param {Account} account
-  */
-  static save(account) {
-    Account.store.add(account);
+  static async changeStatus(accountNumber) {
+    const response = await Account.find('accountnumber', accountNumber);
+    const { status } = response[0];
+    const newStatus = status === 'active' ? 'domant' : 'active';
+    const query = `UPDATE accounts SET status = '${newStatus}' WHERE accountnumber = ${accountNumber} RETURNING status`;
+    const result = await client.query(query);
+    return result.rows[0].status;
   }
 
   /**
   * Delete a given account from datastore
   * @param {string} val
   */
-  static delete(account) {
-    Account.store.delete(account);
+  static async delete(accountnumber) {
+    const query = `DELETE FROM accounts WHERE accountNumber = ${accountnumber}`;
+    const result = await client.query(query);
+    return result;
   }
 
   /**
   * Find a given account number
   * @param {Integer} accountNumber
   * @returns {Account} returns matched account
   */
-  static findByAccountNumber(accountNumber) {
-    return Account.all().find(account => account.accountNumber === parseInt(accountNumber));
+  static async find(column, value) {
+    const query = `SELECT * FROM accounts WHERE ${column} = '${value}'`;
+    const result = await client.query(query);
+    return result.rows;
   }
 
   /**
   * Get all account account numbers
   * @returns {Array} returns all account numbers
   */
-  static all() {
-    return [...Account.store];
+  static async all() {
```

Comment:
Why `static`. Remove it. rename this function to a more descriptive name, maybe `getAllAccountNumber`

Reply:
Noted.